datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/util/OwnerUtils.java

```diff
@@ -150,76 +135,80 @@ private static void addOwner(
             : com.linkedin.common.OwnershipType.CUSTOM;
 
     newOwner.setType(gmsType);
-    newOwner.setTypeUrn(ownershipUrn);
+    newOwner.setTypeUrn(ownershipTypeUrn);
     newOwner.setSource(new OwnershipSource().setType(OwnershipSourceType.MANUAL));
     newOwner.setOwner(ownerUrn);
     ownerArray.add(newOwner);
     ownershipAspect.setOwners(ownerArray);
   }
 
+  private static void removeExistingOwnerIfExists(
+      OwnerArray ownerArray, Urn ownerUrn, Urn ownershipTypeUrn) {
+    ownerArray.removeIf(
+        owner -> {
+          // Remove old ownership if it exists (check ownerUrn + type (entity & deprecated type))
+          return isOwnerEqual(owner, ownerUrn, ownershipTypeUrn);
+        });
+  }
+
+  public static boolean isOwnerEqual(
+      @Nonnull Owner owner, @Nonnull Urn ownerUrn, @Nullable Urn ownershipTypeUrn) {
+    if (!owner.getOwner().equals(ownerUrn)) {
+      return false;
+    }
+    if (owner.getTypeUrn() != null) {
+      return owner.getTypeUrn().equals(ownershipTypeUrn);
+    }
+    if (ownershipTypeUrn == null) {
```

Comment:
I don't think this is right. 2 owners are the same if their urn matches and there isn't a ownershiptype urn? We also need to check for the old ownership type